restlight-core/src/main/java/io/esastack/restlight/core/serialize/BaseHttpBodySerializer.java

```diff
@@ -26,21 +26,21 @@
 public abstract class BaseHttpBodySerializer implements HttpBodySerializer {
 
     @Override
-    public <T> HandledValue<T> deserialize(RequestEntity entity) throws Exception {
+    public <T> Result<T, Void> deserialize(RequestEntity entity) throws Exception {
         if (!supportsRead(entity)) {
-            return HandledValue.failed();
+            return Result.err();
         }
 
-        return HandledValue.succeed(doDeserialize(entity.body().getBytes(), entity.type()));
+        return Result.ok(doDeserialize(entity.body().getBytes(), entity.type()));
     }
 
     @Override
-    public HandledValue<byte[]> serialize(ResponseEntity entity) throws Exception {
+    public Result<byte[], Void> serialize(ResponseEntity entity) throws Exception {
         if (!supportsWrite(entity)) {
-            return HandledValue.failed();
+            return Result.err();
```

Comment:
Err is a more general expression. In fact, if we really want to use a more accurate expression, failed is not very appropriate too. We should express it with unsupported. But for reuse, we reuse result.err(), instead of starting a new stove.in this case, there is no need to be too tangled about whether the expression is completely accurate, and the meaning is roughly the same.